kubectl-plugin/pkg/util/client/client.go

```diff
@@ -108,6 +114,44 @@ func (c *k8sClient) GetRayHeadSvcName(ctx context.Context, namespace string, res
 	}
 }
 
+// WaitRayClusterProvisioned blocks until the RayCluster has a status condition with type=RayClusterProvisioned and status=true.
+// It returns an error if the timeout is reached.
+func (c *k8sClient) WaitRayClusterProvisioned(ctx context.Context, namespace, name string, timeout time.Duration) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	watcher, err := c.RayClient().RayV1().RayClusters(namespace).Watch(ctx, metav1.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("metadata.name", name).String(),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to watch Ray cluster %s in namespace %s: %w", name, namespace, err)
+	}
+	defer watcher.Stop()
+
+	for {
+		select {
+		case <-timeoutCtx.Done():
+			if timeoutCtx.Err() == context.DeadlineExceeded {
+				return fmt.Errorf("timed out waiting for Ray cluster %s in namespace %s to be provisioned", name, namespace)
+			}
+			return ctx.Err()
+		case event := <-watcher.ResultChan():
+			if event.Type == watch.Error {
+				return fmt.Errorf("error watching Ray cluster: %v", event.Object)
+			}
+
+			cluster, ok := event.Object.(*rayv1.RayCluster)
+			if !ok {
+				return fmt.Errorf("unexpected type %T", event.Object)
+			}
+
+			if meta.IsStatusConditionTrue(cluster.Status.Conditions, string(rayv1.RayClusterProvisioned)) {
```

Comment:
```suggestion
			if  cluster.Status.State == rayv1.Ready || meta.IsStatusConditionTrue(cluster.Status.Conditions, string(rayv1.RayClusterProvisioned))
 {
```

And then add tests for this.

Reply:
I added the OR condition but put `cluster.Status.State == rayv1.Ready` second to give conditions precedence.